r/R/util.R

```diff
@@ -86,3 +86,11 @@ all_names <- function(expr) {
 is_constant <- function(expr) {
   length(all_vars(expr)) == 0
 }
+
+handle_embedded_nul_error <- function(e) {
+  msg <- conditionMessage(e)
+  if (grepl(" nul ", msg)) {
+    e$message <- paste0(msg, "; to strip nuls when converting from Arrow to R, set options(arrow.skip_nul = TRUE)")
+  }
+  stop(e)
+}
```

Comment:
Is the call info in the error message at all informative? If not, you might want `stop(msg, call. = FALSE)` inside the conditional (after appending to the message of course).

Reply:
I thought (though I could be remembering wrong) that when you call `stop()` on a caught `simpleError` class object, you just re-raise it and that preserves however the original source of the `stop()` had said `call.` or whatever options. I'll review/confirm tomorrow.

Reply:
Confirmed: 

```
> f <- function(...) stop(...)
> f("boo")
Error in f("boo") : boo
> f("boo", call. = FALSE)
Error: boo
> tryCatch(f("boo"), error = function(e) stop(e))
Error in f("boo") : boo
> tryCatch(f("boo", call. = FALSE), error = function(e) stop(e))
Error: boo
```